transport/src/main/java/io/camunda/zeebe/transport/stream/api/ClientStreamMetrics.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.api;
+
+public interface ClientStreamMetrics {
+  /** Invoked whenever the count of known servers in the registry changes */
+  default void serverCount(final int count) {}
+
+  /** Invoked whenever the count of streams in the registry changes */
+  default void streamCount(final int count) {}
+
+  /** Invoked whenever the count of streams in an aggregated stream changes */
+  default void aggregatedStreamCount(final int count) {}
```

Comment:
Yeah I thought it might be confusing. We can rename them. How about:

`aggregatedStreamCount` -> the number of server streams
`aggregatedClientCount` -> the distribution of client count per server stream

We could also have `clientCount` for the total number of clients.